wsmaster/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/model/impl/AgentImpl.java

```diff
@@ -44,10 +44,10 @@ public AgentImpl(String id,
         this.name = name;
         this.version = version;
         this.description = description;
-        this.dependencies = dependencies;
-        this.properties = properties;
+        this.dependencies = dependencies != null ? dependencies : new ArrayList<>();
```

Comment:
Getters have a bug, I'll rewrite them. The bug is that they return mutable collection that doesn't influence their state.